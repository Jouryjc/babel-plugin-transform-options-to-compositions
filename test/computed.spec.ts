import plugin from '../lib/index'
import { transformSync } from '@babel/core'

describe('computed -> computed', () => {
  test('computed', () => {
    const { code } = transformSync(
      `
      export default {
        computed: {
          name () {
            return this.firstName + this.lastName
          },
      
          secondName: {
            get () {
              return this.lastName + this.firstName
            }
          } 
        },
      }
    `,
      {
        plugins: [plugin],
      }
    )

    expect(code).toMatchSnapshot('computed')
  })
})
