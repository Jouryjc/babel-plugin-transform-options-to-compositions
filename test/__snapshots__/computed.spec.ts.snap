// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computed -> computed computed: computed 1`] = `
"import { computed } from 'vue';
const name = computed(function name() {
  return this.firstName + this.lastName;
});
const secondName = computed({
  get: function get() {
    return this.lastName + this.firstName;
  }
});"
`;
